exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/sequencefile/SequenceFileRecordReader.java

```diff
@@ -0,0 +1,167 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.easy.sequencefile;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+import java.security.PrivilegedExceptionAction;
+
+import com.google.common.base.Stopwatch;
+import org.apache.drill.common.exceptions.DrillRuntimeException;
+import org.apache.drill.common.exceptions.ExecutionSetupException;
+import org.apache.drill.common.expression.SchemaPath;
+import org.apache.drill.common.types.TypeProtos;
+import org.apache.drill.common.types.Types;
+import org.apache.drill.exec.exception.SchemaChangeException;
+import org.apache.drill.exec.ops.OperatorContext;
+import org.apache.drill.exec.physical.impl.OutputMutator;
+import org.apache.drill.exec.record.MaterializedField;
+import org.apache.drill.exec.store.AbstractRecordReader;
+import org.apache.drill.common.types.TypeProtos.MajorType;
+import org.apache.drill.exec.store.dfs.DrillFileSystem;
+import org.apache.drill.exec.util.ImpersonationUtil;
+import org.apache.drill.exec.vector.NullableVarBinaryVector;
+import org.apache.hadoop.io.BytesWritable;
+import org.apache.hadoop.mapred.InputFormat;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.FileSplit;
+import org.apache.hadoop.mapred.Reporter;
+import org.apache.hadoop.mapred.SequenceFileAsBinaryInputFormat;
+import org.apache.hadoop.security.UserGroupInformation;
+
+
+public class SequenceFileRecordReader extends AbstractRecordReader {
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(SequenceFileRecordReader.class);
+
+  private static final int PER_BATCH_RECORD_COUNT = 4096;
+  private static final int PER_BATCH_BYTES = 256*1024;
+
+  private static final MajorType KEY_TYPE = Types.optional(TypeProtos.MinorType.VARBINARY);
+  private static final MajorType VALUE_TYPE = Types.optional(TypeProtos.MinorType.VARBINARY);
+
+  private final SchemaPath keySchema = SchemaPath.getSimplePath("binary_key");
+  private final SchemaPath valueSchema = SchemaPath.getSimplePath("binary_value");
+
+  private NullableVarBinaryVector keyVector;
+  private NullableVarBinaryVector valueVector;
+  private final FileSplit split;
+  private org.apache.hadoop.mapred.RecordReader<BytesWritable, BytesWritable> reader;
+  private final BytesWritable key = new BytesWritable();
+  private final BytesWritable value = new BytesWritable();
+  private final DrillFileSystem dfs;
+  private final String queryUserName;
+  private final String opUserName;
+
+  public SequenceFileRecordReader(final FileSplit split,
+                                  final DrillFileSystem dfs,
+                                  final String queryUserName,
+                                  final String opUserName) {
+    final List<SchemaPath> columns = new ArrayList();
```

Comment:
http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html#ArrayList() 

I don't think we need to put <SchemaPath> around.